Iter9

shilin-anton opened the pull request.

shilin-anton committed:
Add Get and Post handlers with simple storage

shilin-anton committed:
Update responses

shilin-anton committed:
Fix WriteReader order

shilin-anton committed:
Change project structure and add tests

shilin-anton committed:
Close body in a test

shilin-anton committed:
Move body.close before Result

shilin-anton committed:
Add defer on body closing

shilin-anton committed:
Add chi usage, Change tests

shilin-anton committed:
Remove unnecessary storage creation

shilin-anton committed:
Add config to handle flags

shilin-anton committed:
FIx run address

shilin-anton committed:
Fix host

shilin-anton committed:
Add prefix on save short link

shilin-anton committed:
FIx

shilin-anton committed:
Add ENV params

shilin-anton committed:
Fixes after review

shilin-anton committed:
Add logger

shilin-anton committed:
Fix linter offences

shilin-anton committed:
Add /shorten json handler and test

shilin-anton committed:
Fix struct key

shilin-anton committed:
Fix route

shilin-anton committed:
Add gzip middleware

shilin-anton committed:
Add file-manager to control local file storage

Small storage fix

shilin-anton committed:
Fix linter offences

Bazys commented on internal/logger/logger.go:
```diff
@@ -0,0 +1,60 @@
+	}
+
+	// добавляем реализацию http.ResponseWriter
+	loggingResponseWriter struct {
```
не увидел использования этой структуры

shilin-anton replied:
На предыдущих итерациях эти структурки были нужны, просто забил их выпилить. Спасибо за наблюдение

Bazys commented on internal/logger/logger.go:
```diff
@@ -0,0 +1,60 @@
+
+type (
+	// Структура для хранения сведений об ответе
+	responseData struct {
```
так же не увидел использования этой структуры

shilin-anton replied:
здесь та же история, что и с предыдущей структурой)
спасибо

Bazys reviewed (approved):
Сам код написан хорошо, но он написан в нектором структурно-функциональном стиле.
Сейчас ты хранишь только в памяти, дальше надо будет сохранять на диск, а еще дальше - будем работать с БД.
Все время ифчиками это не обложить. 
Посмотри в сторону Dependency Injection и чистой архитектуры.
Вот [ссылка](https://github.com/evrone/go-clean-template?tab=readme-ov-file#dependency-injection) для старта процесса изучения. Но это всего лишь старт, не окончательная версия.
Можно еще [тут](https://quii.gitbook.io/learn-go-with-tests/go-fundamentals/dependency-injection) посмотреть, здесь показано как DI облегчит тестирование в дальнейшем.
Сейчас принимаю, но начинай задумываться над рефакторингом в сторону чистой архитектуры.

shilin-anton committed:
Remove unused types from logger

shilin-anton committed:
Replace storage import by interface in fileManager

shilin-anton committed:
Temp commit

shilin-anton committed:
Refactor code

Remove storage import from file-manager,
Change logic in main,
Change code style in files,
Remove server package,
Fix tests

shilin-anton committed:
More refactor

Separate filemanager layer from handlers,
Improve filemanager handling in storage
